FraudProofs.md

```diff
@@ -0,0 +1,158 @@
+* [Proof of Invalid Oracle Attestation](#proof-of-invalid-oracle-attestation)
+  * [Version 0 `invalid_oracle_attestation_proof`](#version-0-invalid_oracle_attestation_proof)
+* [Proof of Oracle Equivocation](#proof-of-oracle-equivocation)
+  * [Version 0 `oracle_equivocation_proof`](#version-0-oracle_equivocation_proof)
+  * [Version 1 `oracle_equivocation_proof`](#version-1-oracle_equivocation_proof)
+* [Authors](#authors)
+
+## Proof of Oracle Attestation
+
+The fundamental building block of DLC oracle fraud proofs is a proof that a specific oracle
+attested to a specific outcome, which can then be used with other information to prove fraud.
+
+In theory, it would be best if DLC oracles could deliver their announcements and attestations
+using some form of broadcast channel, but in practice it is expected that oracle messages will
+often times be received by users via APIs and other more private and P2P means.
+
+If DLC oracle broadcast channels did exist in such a way that oracles could not commit fraud privately,
+then oracle attestation proofs would be as simple as wrapping the publicly broadcast oracle messages,
+but practically speaking oracles have the ability to cheat without broadcasting fraudulent attestations publicly.
+
+Luckily, enough information is leaked to cheated DLC participants on-chain that they can still
+always generate proofs which show that a fraudulent oracle attestation must exist.
+
+#### Version 0 `oracle_attestation_proof`
+
+1. type: ??? (`oracle_attestation_proof_v0`)
+2. data:
+   * [`u16`:`num_oracles`]
+   * [`num_oracles*oracle_announcement`:`oracle_announcements`]
+   * [`num_oracles*oracle_outcome`:`oracle_outcomes`]
+   * [`32*bytes`:`aggregate_oracle_attestation`]
+   * [`u16`:`oracle_index`]
+3. subtype: `oracle_outcome`
+4. data:
+   * [`nb_signatures*string`:`outcomes`]
+
+The most important piece of this proof is the `aggregate_oracle_attestation` which is recoverable from
+on-chain information as the difference between the broadcast CET's signature and its corresponding adaptor signature.
+In the case that one has access directly to the `oracle_attestation` in question, then this can be used in place of an aggregate.
+
+The `oracle_announcements` and `oracle_outcomes` are used to compute a signature point `S` corresponding to an anticipation of
+these oracles attesting to these outcomes.
+This signature point is used to validate that `aggregate_oracle_attestation*G = S` which proves that there must exist an attestation
+from each of the oracles in `oracle_announcements` of their corresponding outcome in `oracle_outcomes`.
+
+Finally, this proof references a specific oracle by `oracle_index` which is the index in `oracle_announcements` and `oracle_outcomes`
+corresponding to the referenced oracle's announcement and outcome to which they attested. 
+
+## Claim of False Oracle Attestation
+
+The most straightforward kind of oracle fraud occurs when an oracle attests to a false event outcome,
```

Comment:
Really detail, but I feel incorrect would be better than false.

Reply:
I worry about the differentiation between these attestations and "invalid" ones (i.e. not in the announcement) which would both be considered "incorrect" I think? Open to names other than false though

Reply:
Yeah you're right and I cannot come up with something better.